### PR TITLE
Fix #703

### LINKED ISSUE #703: Timestamp-verified restart
A restart function that verifies the restart was successful. Convenient for build scripts. Based on [this example](https://docs.marklogic.com/guide/admin-api/cluster#id_38205).

```
  def verified_restart
    old_timestamp = go(%Q{http://#{@properties["ml.server"]}:8001/admin/v1/timestamp}, "get").body
    restart
    retry_count = 0
    retry_max = 5
    retry_interval= 10
    new_timestamp = old_timestamp
    while retry_count < retry_max do
      begin
        new_timestamp = go(%Q{http://#{@properties["ml.server"]}:8001/admin/v1/timestamp}, "get").body
      rescue
        logger.debug 'Retry attempt ' + retry_count.to_s + ' failed'
      end
      if new_timestamp != old_timestamp
        logger.info 'Verfied restart.'
        break
      end
      logger.debug "Verifying restart..."
      sleep retry_interval
      retry_count += 1
    end
    if new_timestamp == old_timestamp
      logger.warn "Could not verify restart"
    end
  end
end
```

### PR DESCRIPTION
**Pull request by divino**
Note:

We can still skip verification by passing --verify=false

### CODE REVIEW

File: deploy/lib/server_config.rb [hunk 1]
```diff
@@ -556,6 +556,43 @@ def restart
...
   end
 
+  # implemented verified restart
+  def restart
+    verify = find_arg(['--verify'])
+    if verify==='false'
+      restart_basic
+    else
+      # defaults to verified restart
+      old_timestamp = go(%Q{http://#{@properties["ml.server"]}:8001/admin/v1/timestamp}, "get").body
```
Comment (dmcassel) [inline deploy/lib/server_config.rb]: didn't know this endpoint existed. nice!

File: deploy/lib/server_config.rb [hunk 2]
```diff
@@ -556,6 +556,43 @@ def restart
...
+      restart_basic
+      retry_count = 0
+      retry_max = 5
+      retry_interval= 10
```
Comment (dmcassel) [inline deploy/lib/server_config.rb]: I think at least one of these should be property-driven -- desired values may differ depending on the environment. Please add a property to default.properties and make use of it here.

**Comment by divino** [comment]
implemented review points

**Comment by divino** [comment]
Implemented review points. Kindly review again. Thank you.

Dino Bagayaua
Senior Field Engineer
MarkLogic Corporation
Dino.Bagayaua@marklogic.com
Cell: +63-998-569-5332
www.marklogic.com

**Pull request merged**